### PR TITLE
Rework WP source files and update documentation

### PR DESCRIPTION
**Pull request by josesimoes**
- rearrange location of several Wire Protocol source files
- update documentation on Wire Protocol
- minor updates on other documentation files

Signed-off-by: José Simões <jose.simoes@eclo.solutions>

### CODE REVIEW

File: docs/wire-protocol.md [hunk 1]
```diff
@@ -88,8 +92,9 @@ Try to follow this as much as possible when implementing new commands or porting
 Current Wire Protocol implementation has support for transmission over serial port (UART/USART) and serial over USB (USB CDC device class).
 Support for TCP channel is planned at a latter stage.
```
Comment (cw2) [inline docs/wire-protocol.md]: later

File: src/CLR/WireProtocol/WireProtocol_MonitorCommands.c [hunk 2]
```diff
@@ -1,53 +1,22 @@
...
-
-    return true;
-}
-
-
 ////////////////////////////////////////////////////
 
-bool Monitor_Ping(WP_Message* message)
+// provided as weak to be replaced by actual implementation by application
+__attribute__((weak)) bool Monitor_Ping(WP_Message* message)
```
Comment (cw2) [inline src/CLR/WireProtocol/WireProtocol_MonitorCommands.c]: GCC-specific. CLR source should not require a particular compiler/toolchain, i.e. should be compilable with Visual C++ too.

File: src/CLR/WireProtocol/WireProtocol_MonitorCommands.c [hunk 3]
```diff
@@ -1,53 +1,22 @@
...
-    {
-        Monitor_Ping_Reply cmdReply;
-        cmdReply.m_source = Monitor_Ping_c_Ping_Source_NanoBooter;
-
-        ReplyToCommand(message, true, false, &cmdReply, sizeof(cmdReply));
-    }
-
-    return true;
+   // default to false
+   return true;
```
Comment (cw2) [inline src/CLR/WireProtocol/WireProtocol_MonitorCommands.c]: Either the comment or the return value is wrong (contradiction).

File: targets/CMSIS-OS/ChibiOS/nanoCLR/WireProtocol_MonitorCommands.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2017 The nano Framework project contributors
+// Portions Copyright (c) Microsoft Corporation.  All rights reserved.
```
Comment (cw2) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/WireProtocol_MonitorCommands.c]: nanoFramework

File: src/CLR/WireProtocol/WireProtocol_MonitorCommands.c (hunk 2 above)
Comment (josesimoes) [inline src/CLR/WireProtocol/WireProtocol_MonitorCommands.c]: please accept this as it is 
we can change it globally later

**Pull request merged**